modules/docs/docgen/spec/docParser.spec.ts

```diff
@@ -0,0 +1,1419 @@
+    });
+
+    it('should find an external type of a function return type', () => {
+      const program = createProgramFromSource(`
+        export type Foo = (input: number) => Element;
+      `);
+      const docs = parse(program, 'test.ts');
+
+      expect(docs).toHaveProperty('0.name', 'Foo');
+      expect(docs).toHaveProperty('0.type.kind', 'type');
+      expect(docs).toHaveProperty('0.type.value.kind', 'function');
+      expect(docs).toHaveProperty('0.type.value.parameters.0.kind', 'parameter');
+      expect(docs).toHaveProperty('0.type.value.parameters.0.name', 'input');
+      expect(docs).toHaveProperty('0.type.value.parameters.0.type.kind', 'primitive');
+      expect(docs).toHaveProperty('0.type.value.parameters.0.type.value', 'number');
+      expect(docs).toHaveProperty('0.type.value.returnType.kind', 'external');
+    });
+
+    it('should find rest parameters of a function', () => {
+      const program = createProgramFromSource(`
+        export const Foo = (...args: string[]) => null;
+      `);
+      const docs = parse(program, 'test.ts');
+
+      expect(docs).toHaveProperty('0.name', 'Foo');
+      expect(docs).toHaveProperty('0.type.kind', 'function');
+      expect(docs).toHaveProperty('0.type.parameters.0.kind', 'parameter');
+      expect(docs).toHaveProperty('0.type.parameters.0.name', 'args');
+      expect(docs).toHaveProperty('0.type.parameters.0.rest', true);
+      expect(docs).toHaveProperty('0.type.parameters.0.type.kind', 'array');
+      expect(docs).toHaveProperty('0.type.parameters.0.type.value.kind', 'primitive');
+      expect(docs).toHaveProperty('0.type.parameters.0.type.value.value', 'string');
+    });
+
+    it('should find an external type of a function return union type', () => {
+      const program = createProgramFromSource(`
+        export const Foo = (input: {}) => null as Element | null;
+      `);
+      const docs = parse(program, 'test.ts');
+
+      expect(docs).toHaveProperty('0.name', 'Foo');
+      expect(docs).toHaveProperty('0.type.kind', 'function');
+      expect(docs).toHaveProperty('0.type.parameters.0.kind', 'parameter');
+      expect(docs).toHaveProperty('0.type.parameters.0.name', 'input');
+      expect(docs).toHaveProperty('0.type.parameters.0.type.kind', 'object');
+      expect(docs).toHaveProperty('0.type.parameters.0.type.properties', []);
+      expect(docs).toHaveProperty('0.type.returnType.kind', 'union');
+      expect(docs).toHaveProperty('0.type.returnType.value.0.kind', 'external');
+      expect(docs).toHaveProperty('0.type.returnType.value.0.name', 'Element');
+      expect(docs).toHaveProperty(
+        '0.type.returnType.value.0.url',
+        expect.stringContaining('element')
+      );
+      expect(docs).toHaveProperty('0.type.returnType.value.1.kind', 'primitive');
+      expect(docs).toHaveProperty('0.type.returnType.value.1.value', 'null');
+    });
+
+    it('should find an external type of a function return union type', () => {
```

Comment:
This description is for another spec.

Reply:
It is close to the description of another spec. I could only find a single instance of this string.